tests/test_external_mixin.py

```diff
@@ -395,3 +401,141 @@ def test_restore_state(self, strategy):
         assert strategy._connected is False
         assert strategy._use_template_strategy is False
         strategy.state.restore_state.assert_called_once_with(state_dict)
+
+    @parameterized.expand([
+        [ext_strategy_fixture(LoadForecastExternalStrategy())],
+        [ext_strategy_fixture(PVForecastExternalStrategy())]
+    ])
+    def test_set_energy_forecast(self, strategy):
+        transaction_id = str(uuid.uuid4())
+        strategy._get_transaction_id = Mock(return_value=transaction_id)
+        arguments = {"transaction_id": transaction_id,
+                     "energy_forecast": {now().format(DATE_TIME_FORMAT): 1}}
+        payload = {"data": json.dumps(arguments)}
+        assert strategy.pending_requests == deque([])
+        strategy._set_energy_forecast(payload)
+        strategy._get_transaction_id.assert_called_with(payload)
```

Comment:
Nit: it is not necessary to check this (as it is a private method and we should know nothing about it from the "outside").

Reply:
Do you mean `_set_energy_forecast` ? IMO we should test it indeed. It is the callback for setting the energy_forecast.

Reply:
I mean that it's not necessary to test the call to `_get_transaction_id()`